### ISSUE TITLE
APT Repo Returning 403

### ISSUE THREAD

**Issue by sempervictus** [wazuh/wazuh#4410]
Looks like the S3 bucket hosting debs is not doing so well:
```
$ wget --user-agent="Debian APT-HTTP/1.3 (1.6.6)" "https://packages.wazuh.com/3.x/apt/pool/main/w/wazuh-manager/wazuh-manager_3.11.0-1_amd64.deb"
--2020-01-02 03:19:02--  https://packages.wazuh.com/3.x/apt/pool/main/w/wazuh-manager/wazuh-manager_3.11.0-1_amd64.deb
Resolving packages.wazuh.com (packages.wazuh.com)... 13.225.62.46, 13.225.62.11, 13.225.62.81, ...
Connecting to packages.wazuh.com (packages.wazuh.com)|13.225.62.46|:443... connected.
HTTP request sent, awaiting response... 403 Forbidden
2020-01-02 03:19:03 ERROR 403: Forbidden.
```
Tried it with curl, ruby, no dice. Apt itself won't pull 3.11.0 - the containers are u14, so 3.10.2 seems to be the latest they're offered in the repo.
This worked just fine the last few times. S3 limits of some sort?

**Comment by nunofernandes** [comment]
Hi.. I also have the same issue (but with RPMs). I can't download 3.11 (it's not even listed) but I can download 3.10. It was working a few days ago.

![image](https://user-images.githubusercontent.com/1197165/71662318-efa9cf80-2d48-11ea-8e4d-3077bf4cb7a3.png)

**Comment by Theoooooo** [comment]
I also can't downgrade from the 3.11.0 to 3.10.2. Dunno if the repo is the cause but my CentOS can't downgrade because it can't see the x64 version on the package :
```
yum downgrade wazuh-manager
Resolving Dependencies
--> Running transaction check
---> Package wazuh-manager.x86_64 0:3.10.2-1 will be a downgrade
---> Package wazuh-manager.x86_64 0:3.11.0-1 will be erased
--> Finished Dependency Resolution
Error: Package: wazuh-api-3.11.0-1.x86_64 (@wazuh_repo)
           Requires: wazuh-manager = 3.11.0
           Removing: wazuh-manager-3.11.0-1.x86_64 (@wazuh_repo)
               wazuh-manager = 3.11.0-1
           Downgraded By: wazuh-manager-3.10.2-1.x86_64 (wazuh_repo)
               wazuh-manager = 3.10.2-1
           Available: wazuh-manager-3.0.0-1.i386 (wazuh_repo)
               wazuh-manager = 3.0.0-1
           Available: wazuh-manager-3.0.0-2.i386 (wazuh_repo)
               wazuh-manager = 3.0.0-2
           Available: wazuh-manager-3.1.0-1.i386 (wazuh_repo)
               wazuh-manager = 3.1.0-1
           Available: wazuh-manager-3.2.0-1.i386 (wazuh_repo)
               wazuh-manager = 3.2.0-1
           Available: wazuh-manager-3.2.0-2.i386 (wazuh_repo)
               wazuh-manager = 3.2.0-2
           Available: wazuh-manager-3.2.1-1.i386 (wazuh_repo)
               wazuh-manager = 3.2.1-1
           Available: wazuh-manager-3.2.2-1.i386 (wazuh_repo)
               wazuh-manager = 3.2.2-1
           Available: wazuh-manager-3.2.3-1.i386 (wazuh_repo)
               wazuh-manager = 3.2.3-1
           Available: wazuh-manager-3.2.4-1.i386 (wazuh_repo)
               wazuh-manager = 3.2.4-1
           Available: wazuh-manager-3.3.0-1.i386 (wazuh_repo)
               wazuh-manager = 3.3.0-1
           Available: wazuh-manager-3.3.1-1.i386 (wazuh_repo)
               wazuh-manager = 3.3.1-1
           Available: wazuh-manager-3.4.0-1.i386 (wazuh_repo)
               wazuh-manager = 3.4.0-1
           Available: wazuh-manager-3.5.0-1.i386 (wazuh_repo)
               wazuh-manager = 3.5.0-1
           Available: wazuh-manager-3.6.0-1.i386 (wazuh_repo)
               wazuh-manager = 3.6.0-1
           Available: wazuh-manager-3.6.1-1.i386 (wazuh_repo)
               wazuh-manager = 3.6.1-1
           Available: wazuh-manager-3.7.0-1.i386 (wazuh_repo)
               wazuh-manager = 3.7.0-1
           Available: wazuh-manager-3.7.1-1.i386 (wazuh_repo)
               wazuh-manager = 3.7.1-1
           Available: wazuh-manager-3.7.2-1.i386 (wazuh_repo)
               wazuh-manager = 3.7.2-1
           Available: wazuh-manager-3.8.0-1.i386 (wazuh_repo)
               wazuh-manager = 3.8.0-1
           Available: wazuh-manager-3.8.1-1.i386 (wazuh_repo)
               wazuh-manager = 3.8.1-1
           Available: wazuh-manager-3.8.2-1.i386 (wazuh_repo)
               wazuh-manager = 3.8.2-1
           Available: wazuh-manager-3.9.0-1.i386 (wazuh_repo)
               wazuh-manager = 3.9.0-1
           Available: wazuh-manager-3.9.1-1.i386 (wazuh_repo)
               wazuh-manager = 3.9.1-1
           Available: wazuh-manager-3.9.2-1.i386 (wazuh_repo)
               wazuh-manager = 3.9.2-1
           Available: wazuh-manager-3.9.3-1.i386 (wazuh_repo)
               wazuh-manager = 3.9.3-1
           Available: wazuh-manager-3.9.4-1.i386 (wazuh_repo)
               wazuh-manager = 3.9.4-1
           Available: wazuh-manager-3.9.5-1.i386 (wazuh_repo)
               wazuh-manager = 3.9.5-1
           Available: wazuh-manager-3.10.0-1.i386 (wazuh_repo)
               wazuh-manager = 3.10.0-1
           Available: wazuh-manager-3.10.1-1.i386 (wazuh_repo)
               wazuh-manager = 3.10.1-1
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
```

**Comment by alberpilot** [comment]
Hello all
  
  Unfortunately, we undo the 3.11.0 release due to an important issue. More details: https://groups.google.com/forum/#!topic/wazuh/K6IkDAD68po.

  The errors that you are seeing are due to the 3.11.0 undo. Now, the APT and Yum repositories contain the 3.10.2 version as the latest. You both need to update the local repository in order to synchronize with the Wazuh remote repository. for that: 

RPM based OS:
`sudo yum clean all; sudo yum makecache`

The quickest way in Debian based OS: 
`sudo rm /etc/apt/sources.list.d/wazuh.list; sudo apt-get update; sudo echo "deb https://packages.wazuh.com/3.x/apt/ stable main" | tee -a /etc/apt/sources.list.d/wazuh.list; sudo apt-get update`

We will publish 3.11.1 soon, sorry for the inconvenience.

Best regards,
Alberto R

PS: As a note, the Wazuh manager package doesn't support the downgrade. If you don't mind to remove completely the manager and then, to reinstall it just go ahead. Otherwise, if you want to keep data just let me know and we will indicate to you how to manually downgrade the Wazuh manager.

**Comment by Theoooooo** [comment]
@alberpilot I would like to get help on this subject as well as downgrading my agents related to this issue #4192 (which i can't reopen)
The only solution i found is to restart the manager once the CPU start to caps out the usage

**Comment by sempervictus** [comment]
@alberpilot  Why did you pull a tagged release?! Tooling around automation of deployment/configuration expects linear version numbers. A hotfix would have been much more in line with semantic versioning mechanisms.
Any chance of simply pulling out the windows-related commits as a hotfix _while_ this is resolved? We have a clustered deployment now on 3.11 and rolling that back is not feasible for political reasons as much as time allocation required to plan, execute, and validate.

**Comment by Theoooooo** [comment]
@sempervictus I'm in the exact situation as you. I'm rolling back windows agents because i don't if the issue is coming from the manager handling the windows logs or if it's a new format used by the agents. I'll let you know

**Comment by Theoooooo** [comment]
@sempervictus 

As @vikman90 told me, a quick hot fix can be done by reducing the maximum threads consumption for analysisd by the following parameter in "local_internal_options.conf" 
```
echo 'analysisd.winevt_threads=1' >> /var/ossec/etc/local_internal_options.conf
service wazuh-manager restart
```

**Comment by sempervictus** [comment]
@Theoooooo: thanks for the heads up, luckily we're  not _actually seeing this problem_ - but the environment is mostly Linux and OSX, only a handful of windows servers. Our big bug with 3.11 is the fact that vulnerability detection stopped working :(. Not so much a core OSSEC function, but was getting to be a handy tool before it bought the farm.

**Comment by Theoooooo** [comment]
Well as it turns out it only mitigate the CPU usage but not the windows event decoder losses of events. But the issue has been fixed and merged. 
We are only waiting for the push to the repo :D #4412

**Comment by Eowin78** [comment]
Hi,

I'm on debian 9 and I still have the same Issue described on the first post : 

> Err:4 https://packages.wazuh.com/3.x/apt stable/main amd64 wazuh-agent amd64 3.11.0-1
>   403  Forbidden


For your information there is the same issue for rules update : 

> ### Wazuh ruleset ###
> ERROR: Upgrade aborted: Unexpected version in the new ruleset. Expected version v3.10.x. Found version v3.11.0

**Comment by Theoooooo** [comment]
@Eowin78 They didn't update the repos yet. As alberpilot said, they undo the 3.11 from the repo.

Maybe today or tomorrow the changes for the windows_event issue have been merged into the 3.11 branch and you can already build the 3.11.1 version from sources.

**Comment by jeffb255** [comment]
I tried to update to 3.11.1 and received the following:
root@wazuh:/var/ossec/bin# apt update wazuh
E: The update command takes no arguments
root@wazuh:/var/ossec/bin# apt update
Hit:1 http://us.archive.ubuntu.com/ubuntu xenial InRelease
Get:2 http://us.archive.ubuntu.com/ubuntu xenial-updates InRelease [109 kB]
Get:3 http://us.archive.ubuntu.com/ubuntu xenial-backports InRelease [107 kB]
Get:4 http://security.ubuntu.com/ubuntu xenial-security InRelease [109 kB]
Hit:5 https://deb.nodesource.com/node_8.x xenial InRelease
Hit:6 https://artifacts.elastic.co/packages/7.x/apt stable InRelease
Hit:7 https://dl.yarnpkg.com/debian stable InRelease
Fetched 325 kB in 3s (84.7 kB/s)
Reading package lists... Done
Building dependency tree
Reading state information... Done
All packages are up to date.
root@wazuh:/var/ossec/bin# apt upgrade
Reading package lists... Done
Building dependency tree
Reading state information... Done
Calculating upgrade... Done
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
root@wazuh:/var/ossec/bin# apt install wazuh-manager
Reading package lists... Done
Building dependency tree
Reading state information... Done
wazuh-manager is already the newest version (3.11.0-1).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
root@wazuh:/var/ossec/bin# curl -s https://packages.wazuh.com/key/GPG-KEY-WAZUH | apt-key add -
OK
root@wazuh:/var/ossec/bin#
root@wazuh:/var/ossec/bin# echo "deb https://packages.wazuh.com/3.x/apt/ stable main" | tee -a /etc/apt/sources.list.d/wazuh.list
deb https://packages.wazuh.com/3.x/apt/ stable main
root@wazuh:/var/ossec/bin#
root@wazuh:/var/ossec/bin# apt-get update
Hit:1 http://us.archive.ubuntu.com/ubuntu xenial InRelease
Hit:2 http://us.archive.ubuntu.com/ubuntu xenial-updates InRelease
Hit:3 http://us.archive.ubuntu.com/ubuntu xenial-backports InRelease
Get:4 https://packages.wazuh.com/3.x/apt stable InRelease [5,996 B]
Hit:5 http://security.ubuntu.com/ubuntu xenial-security InRelease
Hit:6 https://deb.nodesource.com/node_8.x xenial InRelease
Hit:7 https://artifacts.elastic.co/packages/7.x/apt stable InRelease
Hit:8 https://dl.yarnpkg.com/debian stable InRelease
Get:9 https://packages.wazuh.com/3.x/apt stable/main amd64 Packages [16.3 kB]
Get:10 https://packages.wazuh.com/3.x/apt stable/main i386 Packages [11.2 kB]
Fetched 33.4 kB in 0s (33.7 kB/s)
Reading package lists... Done
root@wazuh:/var/ossec/bin#
root@wazuh:/var/ossec/bin# apt-get update
Hit:1 http://us.archive.ubuntu.com/ubuntu xenial InRelease
Hit:2 http://us.archive.ubuntu.com/ubuntu xenial-updates InRelease
Hit:3 http://security.ubuntu.com/ubuntu xenial-security InRelease
Hit:4 http://us.archive.ubuntu.com/ubuntu xenial-backports InRelease
Hit:5 https://packages.wazuh.com/3.x/apt stable InRelease
Hit:6 https://deb.nodesource.com/node_8.x xenial InRelease
Hit:7 https://artifacts.elastic.co/packages/7.x/apt stable InRelease
Hit:8 https://dl.yarnpkg.com/debian stable InRelease
Reading package lists... Done
root@wazuh:/var/ossec/bin#
root@wazuh:/var/ossec/bin# apt-get install wazuh-manager
Reading package lists... Done
Building dependency tree
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 wazuh-api : Depends: wazuh-manager (< 3.11.1) but 3.11.1-1 is to be installed
E: Error, pkgProblemResolver::Resolve generated breaks, this may be caused by held packages.
root@wazuh:/var/ossec/bin#
root@wazuh:/var/ossec/bin# apt-get install wazuh-api
Reading package lists... Done
Building dependency tree
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 wazuh-api : Depends: wazuh-manager (>= 3.11.1) but 3.11.0-1 is to be installed
E: Unable to correct problems, you have held broken packages.
root@wazuh:/var/ossec/bin# apt-get install wazuh-manager
Reading package lists... Done
Building dependency tree
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 wazuh-api : Depends: wazuh-manager (< 3.11.1) but 3.11.1-1 is to be installed
E: Error, pkgProblemResolver::Resolve generated breaks, this may be caused by held packages.
root@wazuh:/var/ossec/bin# apt-get force
E: Invalid operation force
root@wazuh:/var/ossec/bin# apt-get remove wazuh-manager wazuh-api
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following packages will be REMOVED:
  wazuh-api wazuh-manager
The following held packages will be changed:
  wazuh-api wazuh-manager
0 upgraded, 0 newly installed, 2 to remove and 0 not upgraded.
After this operation, 345 MB disk space will be freed.
Do you want to continue? [Y/n] y
Abort.
root@wazuh:/var/ossec/bin# apt-get remove wazuh-manager wazuh-api
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following packages will be REMOVED:
  wazuh-api wazuh-manager
The following held packages will be changed:
  wazuh-api wazuh-manager
0 upgraded, 0 newly installed, 2 to remove and 0 not upgraded.
After this operation, 345 MB disk space will be freed.
Do you want to continue? [Y/n] y
(Reading database ... 263778 files and directories currently installed.)
Removing wazuh-api (3.11.0-1) ...
Removing wazuh-manager (3.11.0-1) ...
root@wazuh:/var/ossec/bin# apt-get install  wazuh-manager wazuh-api
Reading package lists... Done
Building dependency tree
Reading state information... Done
Suggested packages:
  expect
The following NEW packages will be installed:
  wazuh-api wazuh-manager
0 upgraded, 2 newly installed, 0 to remove and 0 not upgraded.
Need to get 74.2 MB of archives.
After this operation, 345 MB of additional disk space will be used.
Get:1 https://packages.wazuh.com/3.x/apt stable/main amd64 wazuh-manager amd64 3.11.1-1 [72.0 MB]
Get:2 https://packages.wazuh.com/3.x/apt stable/main amd64 wazuh-api amd64 3.11.1-1 [2,192 kB]
Fetched 74.2 MB in 20s (3,551 kB/s)
sh: 0: getcwd() failed: No such file or directory
sh: 0: getcwd() failed: No such file or directory
sh: 0: getcwd() failed: No such file or directory
Selecting previously unselected package wazuh-manager.
(Reading database ... 248751 files and directories currently installed.)
Preparing to unpack .../wazuh-manager_3.11.1-1_amd64.deb ...
Unpacking wazuh-manager (3.11.1-1) ...
Selecting previously unselected package wazuh-api.
Preparing to unpack .../wazuh-api_3.11.1-1_amd64.deb ...
Unpacking wazuh-api (3.11.1-1) ...
Processing triggers for systemd (229-4ubuntu21.23) ...
Processing triggers for ureadahead (0.100.0-19.1) ...
Setting up wazuh-manager (3.11.1-1) ...
Setting up wazuh-api (3.11.1-1) ...
cp: cannot stat '/var/ossec/packages_files/api_config_files/api/configuration/auth/user': No such file or directory
dpkg: error processing package wazuh-api (--configure):
 subprocess installed post-installation script returned error exit status 1
sh: 0: getcwd() failed: No such file or directory
                                                 sh: 0: getcwd() failed: No such file or directory
                                                                                                  E: Sub-process /usr/bin/dpkg returned an error code (1)

**Comment by Theoooooo** [comment]
The update is now available to the officiel CentOS Repo.

I think this issue can be closed.

**Comment by BraulioV** [comment]
Hi @jeffb255,

TL;DR: you need to fully uninstall the manager and the api by running: `apt purge wazuh-manager wazuh-api` and install them again with `apt install wazuh-manager wazuh-api`.

For more details, the error `cp: cannot stat '/var/ossec/packages_files/api_config_files/api/configuration/auth/user': No such file or directory` was an unkown bug that appears if Wazuh API was configured by the script `/var/ossec/api/scripts/configure_api.sh`, then it is uninstalled with `apt remove` and finally, it is installed again with `apt install`. 

The API's user files are removed while running `apt remove wazuh-api`, so, when the package tries to restore them in the post-install process, the package fails because it can't find those files. This error has been solved in this PR https://github.com/wazuh/wazuh-packages/issues/375 and it will be released with the next version of Wazuh. 

To solve this issue, you will need to completely remove the wazuh-api package, upgrade the manager to v3.11.1 and finally, install the v3.11.1 wazuh-api package. 

The error: `sh: 0: getcwd() failed: No such file or directory` is caused by running a command from a directory that no longer exists. That directory is `/var/ossec/bin` in your case. Since you are installing and uninstalling Wazuh packages in this directory when the package is removed, that directory will no longer exist and the shell won't be able to get the current directory work. Try to run `cd $HOME` and run again the commands. That will remove this error.

I hope it helps.

**Comment by sempervictus** [comment]
With the repos fixed and carrying new versions, this can be closed, but the broken vuln detector thing persists even on 3.11.1

**Comment by vikman90** [comment]
@sempervictus Could you give us more information about your issue with Vulnerability Detector? Do you mean that _wazuh-modulesd_ is crashing?

That may be due to a memory usage peak that makes Linux kill the process on systems with less than 1.5 GiB RAM (issue #4398, fixed by #4427).

Best regards.
